### PR TITLE
fix(file): keep ignored .gitignore rules active

### LINKED ISSUE #624: Behavior of --ignore "**/.gitignore"
### Description

Thank you for the great project.

The behavior of --ignore "**/.gitignore" seems not correct.

I just want to exclude the content of .gitignore from the output, but I want to keep using filters listed in .gitignore

`repomix --ignore "**/.gitignore"` excludes .gitignore, but also disable the filters described in the .gitignore

### Usage Context

None

### Repomix Version

_No response_

### Node.js Version

_No response_

### PR DESCRIPTION
**Pull request by Samsen879**
## Summary

Fixes #624.

This PR fixes a conflict where ignore-control files such as `.gitignore`, `.ignore`, and `.repomixignore` can be ignored before globby has a chance to load their rules.

The fix keeps these control files visible during traversal so their rules are applied, then filters the control files out of the final file list.

## Why

If users configure custom ignore patterns such as `**/.gitignore`, the `.gitignore` file itself should not appear in Repomix output. However, hiding it too early can also prevent its rules from being respected.

Expected behavior:

- `.gitignore` / `.ignore` / `.repomixignore` are not included in output
- rules inside those files are still applied
- existing custom/default ignore behavior is preserved

## Tests

- [x] `npm test tests/core/file/fileSearch.gitignoreSpec.test.ts`
- [x] `npm test tests/core/file/dotIgnoreSpec.test.ts`
- [x] `npm test tests/core/file/fileSearch.test.ts`
- [x] `npm run lint`
- [x] `npm run build`
- [x] `npm test`

Note: `npm run lint` exits 0, but upstream main still reports 3 existing warnings unrelated to this change.

### CODE REVIEW

File: src/core/file/fileSearch.ts [hunk 1]
```diff
@@ -86,6 +88,26 @@ export const normalizeGlobPattern = (pattern: string): string => {
...
+
+const filterDeferredIgnoredFiles = (filePaths: string[], deferredIgnorePatterns: string[]): string[] => {
+  if (deferredIgnorePatterns.length === 0) {
+    return filePaths;
+  }
+  return filePaths.filter((filePath) => {
+    const normalizedPath = toPosixPath(filePath);
+    return !deferredIgnorePatterns.some((pattern) => minimatch(normalizedPath, toPosixPath(pattern), { dot: true }));
+  });
+};
```
[the remark this answers is not in the document]
Comment (yamadashy) [reply src/core/file/fileSearch.ts:124]: No action needed — `posixPatterns` is already pre-computed once via `.map()` before the `filePaths.filter()` loop, so `toPosixPath` is not called per file/pattern. Note the current code maps with `toPosixIgnorePattern` (not raw `toPosixPath`) to also strip trailing slashes; applying the suggestion verbatim would regress that fix. 🤖

**Review by yamadashy — APPROVED** [review]

**Comment by yamadashy** [comment]
Thanks for the PR, really nice fix for #624! 🙏

I pushed a couple of small follow-up commits (with tests) covering two edge cases: a trailing-slash variant (`**/.gitignore/`), and a directory literally named `.gitignore`.

Everything looks good on my end, so I'm merging this. Thanks again!
